File: CoreJava/game-tic-tac-toe-swing/src/com/techlab/game/ResultAnalyser.java
package com.techlab.game;

import javax.swing.JButton;

public class ResultAnalyser {

	JButton cell[];

	public ResultAnalyser(JButton[] cell) {
		this.cell = cell;
	}

	private boolean checkVertical() {
		int i, j;
		for (i = 0; i <= Board.getBoardSize() * (Board.getBoardSize() - 1); i = i + Board.getBoardSize()) {
			for (j = i; j < i + (Board.getBoardSize() - 1); j++) {
				if (cell[i].getText().isEmpty())
					return false;
				else {
					if (!cell[j].getText().equals(cell[j + 1].getText()))
						break;
				}

			}
			if (j == i + (Board.getBoardSize() - 1)) {
				return true;
			}
		}
		return false;
	}

	private boolean checkHorizontal() throws Exception{

		int i, j;
		for (i = 0; i < Board.getBoardSize(); i++) {
			for (j = i; j < i + (Board.getBoardSize() * (Board.getBoardSize() - 1)); j = j + Board.getBoardSize()) {

				System.out.println("j======" + j + "-----------" + Cell.cell[j].getText());

				String one = Cell.cell[j].getText();
				String two = Cell.cell[j + Board.getBoardSize()].getText();

				if (one.equals(null))
					break;

				if (!one.equals(two))
					break;

			}
			if (j == i + (Board.getBoardSize() * (Board.getBoardSize() - 1))) {
				return true;
			}
		}
		return (false);

	}

	private boolean checkDiagonally() {

		int i = 0;
		for (i = Board.getBoardSize() - 1; i < Board.getBoardSize()
				* (Board.getBoardSize() - 1); i += Board.getBoardSize() - 1) {

			if (cell[i].getText().equals(null))
				return false;
			else {
				if (!cell[i].getText().equals(cell[i + Board.getBoardSize() - 1].getText()))
					break;
			}

		}

		if (i == Board.getBoardSize() * (Board.getBoardSize() - 1))
			return true;
		return false;

	}

	private boolean checkAntiDiagonally() {

		int i = 0;
		for (i = 0; i < (Board.getBoardSize() * Board.getBoardSize()) - 1; i += Board.getBoardSize() + 1) {

			if (cell[i].getText().isEmpty())
				return false;
			else {
				if (!cell[i].getText().equals(cell[i + Board.getBoardSize() + 1].getText()))
					break;
			}

		}

		if (i == (Board.getBoardSize() * Board.getBoardSize()) - 1)
			return true;
		return false;

	}

	public boolean checkWin() throws Exception {
//		return (checkHorizontal() || checkVertical() || checkDiagonally() || checkAntiDiagonally());
		return  checkHorizontal();
	}

}

File: C#Folder/TicTacToeGameSolution/TicTacToeGameLib/ResultAnalyser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace TicTacToeGameLib
{
    public class ResultAnalyser : Form
    {
        Button[] cell;

        public ResultAnalyser(Button[] cell)
        {
            this.cell = cell;
        }

        private bool checkVertical()
        {
            int i, j;
            for (i = 0; i <= Board.BoardSize * (Board.BoardSize - 1); i = i + Board.BoardSize)
            {
                for (j = i; j < i + (Board.BoardSize - 1); j++)
                {
                    if (cell[i].Text == "")
                        return false;
                    else
                    {
                        if (!cell[j].Text.Equals(cell[j + 1].Text))
                            break;
                    }

                }
                if (j == i + (Board.BoardSize - 1))
                {
                    return true;
                }
            }
            return false;
        }

        private bool checkHorizontal()
        {

            int i, j;
            for (i = 0; i < Board.BoardSize; i++)
            {
                for (j = i; j < i + (Board.BoardSize * (Board.BoardSize - 1)); j = j + Board.BoardSize)
                {

                    Console.WriteLine("j======" + j + "-----------" + Cell.cell[j].Text);

                    String one = Cell.cell[j].Text;
                    String two = Cell.cell[j + Board.BoardSize].Text;

                    if (one.Equals(""))
                        break;

                    if (!one.Equals(two))
                        break;

                }
                if (j == i + (Board.BoardSize * (Board.BoardSize - 1)))
                {
                    return true;
                }
            }
            return (false);

        }

        private bool checkDiagonally()
        {

            int i = 0;
            for (i = Board.BoardSize - 1; i < Board.BoardSize
                    * (Board.BoardSize - 1); i += Board.BoardSize - 1)
            {

                if (cell[i].Text.Equals(""))
                    return false;
                else
                {
                    if (!cell[i].Text.Equals(cell[i + Board.BoardSize - 1].Text))
                        break;
                }

            }

            if (i == Board.BoardSize * (Board.BoardSize - 1))
                return true;
            return false;

        }

        private bool checkAntiDiagonally()
        {

            int i = 0;
            for (i = 0; i < (Board.BoardSize * Board.BoardSize) - 1; i += Board.BoardSize + 1)
            {

                if (cell[i].Text == "")
                    return false;
                else
                {
                    if (!cell[i].Text.Equals(cell[i + Board.BoardSize + 1].Text))
                        break;
                }

            }

            if (i == (Board.BoardSize * Board.BoardSize) - 1)
                return true;
            return false;

        }

        public bool checkWin()
        {
            return (checkHorizontal() || checkVertical() || checkDiagonally() || checkAntiDiagonally());
            // return  checkHorizontal();
        }
    }
}
